Sign_Up_Demo/Controllers/CustomPideBienViewController.swift

```diff
@@ -0,0 +1,120 @@
+                return self?.setupVerticalScrollSection()
+            case .none:
+                fatalError("Should not be none")
+            }
+        }
+        return layout
+    }()
+    
+    // MARK: - Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpCollectionView()
+    }
+    
+    // MARK: - Private Methods
+    private func setUpCollectionView() {
+        collectionView.register(HorizontalCollectionCell.nib(), forCellWithReuseIdentifier: HorizontalCollectionCell.reuseIdentifier)
+        collectionView.register(VerticalCollectionCell.nib(), forCellWithReuseIdentifier: VerticalCollectionCell.reuseIdentifier)
+        collectionView.register(HeaderCollectionReusableView.nibForHeaterAndFooter(), forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: HeaderCollectionReusableView.reuseIdentifierForHeaterAndFooter)
+        collectionView.register(FooterCollectionReuseableView.nibForHeaterAndFooter(), forSupplementaryViewOfKind: UICollectionView.elementKindSectionFooter, withReuseIdentifier: FooterCollectionReuseableView.reuseIdentifierForHeaterAndFooter)
+        collectionView.collectionViewLayout = compositionalLayout
+        collectionView.dataSource = self
+        collectionView.contentInsetAdjustmentBehavior = .scrollableAxes
+    }
+    
+    // MARK: - Internal Methods
+    func setupHorizontalScrollSection() -> NSCollectionLayoutSection {
+        let item = NSCollectionLayoutItem(
+            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0), heightDimension: .fractionalHeight(1.0)))
+        let group = NSCollectionLayoutGroup.horizontal(
+            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.90), heightDimension: .absolute(204)), subitem: item, count: 1)
+        let section = NSCollectionLayoutSection(group: group)
+        section.interGroupSpacing = 8
+        section.contentInsets = NSDirectionalEdgeInsets(top: 125.0, leading: 16.0, bottom: -15.0, trailing: 16.0)
+        section.orthogonalScrollingBehavior = .groupPaging
+        return section
+    }
+    
+    func setupVerticalScrollSection() -> NSCollectionLayoutSection {
+        let item = NSCollectionLayoutItem(
+            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.5), heightDimension: .fractionalWidth(0.5)))
+        item.contentInsets = NSDirectionalEdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 15)
+        let group = NSCollectionLayoutGroup.horizontal(
+            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0), heightDimension: .absolute(165)), subitem: item, count: 2)
+        group.contentInsets = NSDirectionalEdgeInsets(top: 0.0, leading: 16.0, bottom: 0.0, trailing: 1.0)
+        let section = NSCollectionLayoutSection(group: group)
+        section.interGroupSpacing = 20
+        let headerView = NSCollectionLayoutBoundarySupplementaryItem(
```

Comment:
Usually type names start with NS are objective-C classes but for most of them there are corresponding Swift structs like String for NSString etc. Check and use if there are Swift's corresponding type for them.

Reply:
Najeeb I think it is a general way to create composition collection layout. There is no specific structs for swifts.